internal/reconciler/logpipeline/status.go

```diff
@@ -61,24 +62,23 @@ func (r *Reconciler) updateStatusUnsupportedMode(ctx context.Context, pipeline *
 }
 
 func (r *Reconciler) setAgentHealthyCondition(ctx context.Context, pipeline *telemetryv1alpha1.LogPipeline) {
```

Comment:
This code looks quite similar to the one here: internal/reconciler/logparser/status.go:40
we should make this a reusable function